File: SuperiorPackGroup/BLL/PoolBomBLL.vb
Imports DevExpress.Data.Filtering
Imports DXDAL.SPGData
Imports DevExpress.Xpo
Imports System.Text

Public Class PoolBomBLL

    Public Shared Function PoolUsedAsBom(ByVal poolID As Integer) As Boolean

        Using lXPCollection As XPCollection(Of PoolBom) = New XPCollection(Of PoolBom)(Session.DefaultSession, New BinaryOperator(PoolBom.Fields.ItemPoolID.PropertyName, poolID, BinaryOperatorType.Equal))
            Return lXPCollection.Count > 0
        End Using

    End Function

End Class

File: SuperiorPackGroup/BLL/PoolBomBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

using DevExpress.Data.Filtering;
using DXDAL.SPGData;
using DevExpress.Xpo;
using System.Text;

namespace SuperiorPackGroup
{
	public class PoolBomBLL
	{

		public static bool PoolUsedAsBom(int poolID)
		{

			using (XPCollection<PoolBom> lXPCollection = new XPCollection<PoolBom>(Session.DefaultSession, new BinaryOperator(PoolBom.Fields.ItemPoolID.PropertyName, poolID, BinaryOperatorType.Equal)))
			{
				return lXPCollection.Count > 0;
			}

		}

	}

}